ch5-catdir/utils.cpp

```diff
@@ -0,0 +1,58 @@
+#include "utils.h"
+
+#include <cctype>
+
+#include <sys/stat.h>
+
+namespace
+{
+
+auto typeChar(mode_t mode)
+{
+    if (S_ISREG(mode))  { return '-'; }
+    if (S_ISDIR(mode))  { return 'd'; }
+    if (S_ISLNK(mode))  { return 'l'; }
+    if (S_ISCHR(mode))  { return 'c'; }
+    if (S_ISBLK(mode))  { return 'b'; }
+    if (S_ISFIFO(mode)) { return 'p'; }
+    if (S_ISSOCK(mode)) { return 's'; }
+    return '?';
```

Comment:
Maybe throw an exception instead?

Reply:
POSIX says that implementations may have additional file types, and I don't want my application to fail on such systems.

Reply:
Interesting! Thanks